### PR TITLE
finish porting the radiation consup + advect_fspace to C++

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
relies on PR #1188
fixes PR #1184

**Comment by maxpkatz** [comment]
Can we look at the CPU and GPU test suite results?

**Comment by zingale** [comment]
multigroup CPU test problems: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-09-08-001/index.html

**Comment by zingale** [comment]
GPU tests: http://groot.astro.sunysb.edu/Castro/test-suite/gpu/2020-09-15-003/index.html

File: Source/radiation/RadHydro.H [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+dudt(Real* u, Real* a, const Real* dx, const int n, Real* dudt_tmp) {
+
+  // compute the RHS for the g'th energy group out of n note, since
+  // this is advection in frequency space, we will need to assume some
+  // boundary conditions on the stencil used to reconstruct over nu
+
+  Real f[NGROUPS+1];
+
+  // we need 2 ghost cells on each end
+  Real ag[NGROUPS+4];
```
Comment (maxpkatz) [inline Source/radiation/RadHydro.H:109]: Perhaps we should use Array1D here so we can retain the indexing from Fortran?

**Comment by zingale** [comment]
all tests now pass with the Array1D:
http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-11-02-003/index.html

turn out i was reflecting the ug BCs when they should have been outflow

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# 20.12
+
+   * The radiation frequency space advection was converted to C++ (#1189)
```
Comment (maxpkatz) [inline CHANGES.md:3]: I don't have any problem with this but just FYI my strategy has been to only update this file for user-facing/problem setup-related changes, not for internal implementation details.

File: Source/radiation/RadHydro.H (hunk 1 above)
Comment (zingale) [reply Source/radiation/RadHydro.H]: done

File: CHANGES.md (hunk 2 above)
Comment (zingale) [reply CHANGES.md]: removed

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**